### ISSUE TITLE
Messages related issues (in dev.app) - 4

### ISSUE THREAD

**Issue by madhurrya** [sartography/spiff-arena#1800]
1. When a non admin user clicks on a message task, it gives this error

![Image](https://github.com/sartography/spiff-arena/assets/110527567/4bd50948-c8a7-4b53-a6b3-f9bc79eaa0b9)

![Image](https://github.com/sartography/spiff-arena/assets/110527567/ed1de53b-3716-4c05-b02c-0010a9f3b3a8)

**Comment by madhurrya** [comment]
2. In the send messages, send data we show the sent data when we click 'View" like this.
But in the Receive messages, the receive data is empty. Shouldn't it show the same there?

https://dev.app.spiff.status.im/i/3057
![image](https://github.com/sartography/spiff-arena/assets/110527567/d819fec5-8d5b-4843-95a5-223c10e67adf)

![image](https://github.com/sartography/spiff-arena/assets/110527567/e4c6ac57-f318-4a67-ae0f-ec059d7a6573)


https://dev.app.spiff.status.im/i/3058

![image](https://github.com/sartography/spiff-arena/assets/110527567/39d246a8-137e-4b32-97af-bb8ad8d65ad5)

**Comment by burnettk** [comment]
@madhurrya this might be another issue where the correlation name must not contain spaces. that might be something to try to make sure it acts the same way when there are no spaces.

**Comment by madhurrya** [comment]
@burnettk tested it without adding a matching condition. In that case the receive data is empty.
Eventhough a matching condition is not set, it should still show the receive data right?

https://dev.app.spiff.status.im/i/3285
https://dev.app.spiff.status.im/i/3286

![image](https://github.com/sartography/spiff-arena/assets/110527567/1563ac52-a80c-4404-879c-2be088ca39cf)

![image](https://github.com/sartography/spiff-arena/assets/110527567/1a38488d-02b4-46d9-a78f-5f4a447317b3)

**Comment by madhurrya** [comment]
And if you take this one here it shows Receive data even before the message completes. May be I have not understood the concept there?
https://dev.app.spiff.status.im/i/3291
![image](https://github.com/sartography/spiff-arena/assets/110527567/24e17a53-7fa8-4295-baed-8a2e21d3fe9f)

![image](https://github.com/sartography/spiff-arena/assets/110527567/1b0bc6f7-0ade-4432-9b7a-e86de99f85e6)

**Comment by burnettk** [comment]
maybe because of matching on empty correlations. we will take a look at this.

**Comment by madhurrya** [comment]
Issue 1 is fixed.
Issue 2 seems to be there for this instance - https://dev.app.spiff.status.im/i/4428
![image](https://github.com/sartography/spiff-arena/assets/110527567/94e008e1-9333-4158-af8e-297faef30f1c)

**Comment by jasquat** [comment]
I think the new issue is slightly different and more interesting. The message receive task was started manually, not from a message start event, and was never given any correlation keys. This means its keys are empty and therefore it will receive any sender task for that same message.

I've always wondered if it should act like this. The problem being that a message that was mistakenly never given any keys could hijack ongoing conversations between other process instances.

**Comment by madhurrya** [comment]
It has the correlation properties defined. 

![image](https://github.com/sartography/spiff-arena/assets/110527567/2ad07d76-5437-4812-8fa5-7d220e36a6ea)


This is what we see in the throw event.

![image](https://github.com/sartography/spiff-arena/assets/110527567/016aa769-66fd-4403-86f7-75aaddf5891e)

**Comment by jasquat** [comment]
Yeah the properties but it was never given any values to those properties prior to running. Which is why it's a blank dict on the receive.

**Comment by madhurrya** [comment]
Do you mean a matching condition?
From the sending side it's sending a value. I didn't get what you mean. How do we set values on the receiving side?
